### PR TITLE
build: Upgrading dependencies for compatibility with Node 20 and general maintenance

### LINKED ISSUE #14: Update dependencies to work on Node 20
Node 16 builds will be deprecated by Vercel on **February 5, 2024**.

### PR DESCRIPTION
**Pull request by msohaill**
### List of changes:

- Upgrading all dependencies to latest stable versions.
- No major API adjustments required apart from:
  - Swiper CSS filename changes ([docs](https://swiperjs.com/swiper-api#css-styles)). I have not been able to test this out yet due to (I assume) no mentor/person data.
  - Props for styled-components. See [here](https://gist.github.com/meddokss/009cdb5192e487654c0d90cd8f692214).

### Type of change:

- [ ] Bug fix
- [ ] New feature
- [x] Breaking change
- [ ] New release
- [ ] Requires a documentation update

### How did you do this?
Basically just using `ncu -u` (npm-check-upgrade) to update all dependencies and then updating system Node version and then doing some testing/tinkering.

### Questions for code reviewers?
Is there a way to test out the Swiper Carousel/Navigation component?

### PR Checklist:

- [ ] Merged `develop` branch (before testing)
- [ ] Ran `yarn format` to format code
- [ ] Listed change(s) in the Changelog
- [ ] Tested all links in project relevant browsers
- [ ] Tested all links on different screen sizes
- [ ] Referenced all useful info (issues, tasks, etc)

### CODE REVIEW

**Comment by jackyyzhang03** [comment]
> Is there a way to test out the Swiper Carousel/Navigation component?

You can make a copy of the Static Site Content Google Sheet and change the `PEOPLE_GOOGLE_SPREADSHEET_IDENTIFIER` environment variable. Note that the service account needs to be able to view the Google Sheet. Then you can add some people which will show up in the carousel.

**Comment by jackyyzhang03** [comment]
Running `gatsby build` fails, seems to have to do with `swiper` package?

**Comment by msohaill** [comment]
> Running `gatsby build` fails, seems to have to do with `swiper` package?

Will take a look. I don't think I have access to the Vercel thing so couldn't tell what the issue was 😂

**Comment by msohaill** [comment]
> Will take a look. I don't think I have access to the Vercel thing so couldn't tell what the issue was 😂
Seems the error was to do with JSX variables and such. Adjusted the interpolation and all is good now. Also tested the swiper carousels and everything looks great!


The other build error was due to an ill-formatted Static Site Content spreadsheet (empty row). Deleted the row from the spreadsheet and all is good!

**Review by jackyyzhang03 — APPROVED** [review]
LGTM
